src/com/amazon/ionelement/api/IonOperators.kt

```diff
@@ -0,0 +1,410 @@
+@IonOperators
+public infix fun DE.gte(other: DE): Boolean = this.decimalValue >= other.decimalValue
+
+/* DecimalElement with BigDecimal operators */
+@IonOperators
+public operator fun DE.plus(other: BigDecimal): DE = ionDecimal(Decimal.valueOf(this.decimalValue + other), this.annotations, this.metas)
+@IonOperators
+public operator fun DE.minus(other: BigDecimal): DE = ionDecimal(Decimal.valueOf(this.decimalValue - other), this.annotations, this.metas)
+@IonOperators
+public operator fun DE.times(other: BigDecimal): DE = ionDecimal(Decimal.valueOf(this.decimalValue * other), this.annotations, this.metas)
+@IonOperators
+public operator fun DE.div(other: BigDecimal): DE = ionDecimal(Decimal.valueOf(this.decimalValue / other), this.annotations, this.metas)
+@IonOperators
+public operator fun DE.rem(other: BigDecimal): DE = ionDecimal(Decimal.valueOf(this.decimalValue % other), this.annotations, this.metas)
+@IonOperators
+public infix fun DE.eq(other: BigDecimal): Boolean = this.decimalValue == other
+@IonOperators
+public infix fun DE.lt(other: BigDecimal): Boolean = this.decimalValue < other
+@IonOperators
+public infix fun DE.gt(other: BigDecimal): Boolean = this.decimalValue > other
+@IonOperators
+public infix fun DE.lte(other: BigDecimal): Boolean = this.decimalValue <= other
+@IonOperators
+public infix fun DE.gte(other: BigDecimal): Boolean = this.decimalValue >= other
+
+/* BigDecimal with DecimalElement operators */
+@IonOperators
+public operator fun BigDecimal.plus(other: DE): DE = ionDecimal(Decimal.valueOf(this + other.decimalValue), other.annotations, other.metas)
+@IonOperators
+public operator fun BigDecimal.minus(other: DE): DE = ionDecimal(Decimal.valueOf(this - other.decimalValue), other.annotations, other.metas)
+@IonOperators
+public operator fun BigDecimal.times(other: DE): DE = ionDecimal(Decimal.valueOf(this * other.decimalValue), other.annotations, other.metas)
+@IonOperators
+public operator fun BigDecimal.div(other: DE): DE = ionDecimal(Decimal.valueOf(this / other.decimalValue), other.annotations, other.metas)
+@IonOperators
+public operator fun BigDecimal.rem(other: DE): DE = ionDecimal(Decimal.valueOf(this % other.decimalValue), other.annotations, other.metas)
+@IonOperators
+public infix fun BigDecimal.eq(other: DE): Boolean = this == other.decimalValue
+@IonOperators
+public infix fun BigDecimal.lt(other: DE): Boolean = this < other.decimalValue
+@IonOperators
+public infix fun BigDecimal.gt(other: DE): Boolean = this > other.decimalValue
+@IonOperators
+public infix fun BigDecimal.lte(other: DE): Boolean = this <= other.decimalValue
+@IonOperators
+public infix fun BigDecimal.gte(other: DE): Boolean = this >= other.decimalValue
+
+/* FloatElement unary operators */
+@IonOperators
+public operator fun FE.unaryMinus(): FE = ionFloat(-this.doubleValue, this.annotations, this.metas)
+@IonOperators
+public operator fun FE.unaryPlus(): FE = ionFloat(+this.doubleValue, this.annotations, this.metas)
```

Comment:
I don't think it's worth implementing this function (or any other `unaryPlus` functions). It's the identity function when applied to any number, so it's a function that does nothing useful, but it does come at a cost (i.e. it allocates a new `FloatElement` instance). No one should want to use it, so we shouldn't provide it.